### PR TITLE
Larger mesh support

### PR DESCRIPTION
**Pull request by MartinKarp**
This fixes issues with generating nmsh files for meshes with more than 8 or 9 Million elements.

### CODE REVIEW

File: src/io/nmsh_file.f90 [hunk 1]
```diff
@@ -44,7 +44,9 @@ module nmsh_file
   implicit none
   
   private
-
+  !> Specifices the maximum number of elements any rank is allowed to write (for nmsh).
+  !! Needed in order to generate large meshes where an individual write might exceed 2GB.
+  integer :: max_write_nel = 8000000
```
Comment (njansson) [inline src/io/nmsh_file.f90:49]:
```suggestion
  integer, parameter :: max_write_nel = 8000000 
```

File: src/io/re2_file.f90 [hunk 2]
```diff
@@ -417,7 +417,7 @@ subroutine re2_file_read_curve(msh, ncurve, dist, fh, mpi_offset, v2_format)
           end do
        end do
     end do
-  
+    write(*,*) 'reading curved data'
```
Comment (njansson) [inline src/io/re2_file.f90:420]: Can't we do this with neko_log

**Review by njansson — CHANGES_REQUESTED** [review]

File: src/io/re2_file.f90 (hunk 2 above)
Comment (MartinKarp) [reply src/io/re2_file.f90:420]: We use write everywhere else in re2

Comment (MartinKarp) [reply src/io/re2_file.f90:420]: Maybe they should all be changed

Comment (njansson) [reply src/io/re2_file.f90:420]: Ok, should probably change, but in a different pr of course

**Review by njansson — APPROVED** [review]

**Pull request merged**